### PR TITLE
create navbar compnent

### PR DESCRIPTION
**Pull request by MohammadAlHabil**
relates #19

image
image

### CODE REVIEW

File: client/src/App/index.jsx [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+        setCategories(res.data);
+      });
+
+    return () => cancelTokenSource.cancel();
+  }, [categories]);
+
   return (
-    <div>Clean Stead</div>
+    <Router>
+      <ConfigProvider locale={ArEg}>
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:27]: Move this Provider to the `index.js`

**Review by MostafaQanoo — APPROVED** [review]

File: client/src/Components/Navbar/index.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        {
+          label: user.role === 'admin'
+            ? (<Link to="/dashboard">Dashboard</Link>)
+            : (<Link to="/my-appointment">My Appointment</Link>)
+          ,
+        },
+        {
+          label: (
+            <Button className="logout" onClick={logout}>LogOut</Button>
+          ),
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/index.jsx:60]:
```suggestion
          label: 'Logout',
          onClick: logout,
```

File: client/src/Components/Navbar/index.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      <Image preview={false} src={logo} width={150} />
+      <Menu mode="horizontal" defaultSelectedKeys={NavItems[0]} items={NavItems} />
+      <div>
+        {isLogged ? (
+          <Space direction="vertical">
+            <Dropdown overlay={avatarMenu} placement="bottom">
+              <Space>
+                <div className="avatar">
+                  <Avatar size={32} icon={<UserOutlined />} />
+                  {user.name}
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/index.jsx:77]:
```suggestion
                  &nbsp;{user.name}&nbsp;
```

File: client/src/App/index.jsx [hunk 4]
```diff
@@ -11,26 +11,28 @@ function App() {
...
 
     return () => cancelTokenSource.cancel();
   }, [categories]);
 
   return (
     <Router>
       <Layout>
-        <Navbar isLogged={false} categories={categories} user={{ name: 'Mohammad', role: 'admin' }} />
+        <Navbar
+          isLogged={true}
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:29]:
```suggestion
          isLogged
```

File: client/src/App/index.jsx [hunk 5]
```diff
@@ -11,26 +11,28 @@ function App() {
...
+          categories={categories}
+          user={{ name: 'Mohammad', role: 'admin' }}
+        />
         <Header />
         <Routes>
-          <Route
-            path="/"
-            element={<div>Home</div>}
-          />
+          <Route path="/" element={<div>Home</div>} />
```
Comment (Mu7ammadAbed) [inline client/src/App/index.jsx:35]:
```suggestion
```

File: client/src/Components/Navbar/index.jsx [hunk 6]
```diff
@@ -39,25 +43,25 @@ function Navbar({ isLogged, categories, user }) {
   }));
 
   const logout = () => {
-    axios.get('/logout')
-      .then((res) => {
-        console.log(res.data);
-      });
+    axios.get('/logout').then((res) => {
+      console.log(res.data);
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/index.jsx:47]:
```suggestion
      console.log('Logged out');
```

File: client/src/Components/Navbar/index.jsx [hunk 7]
```diff
@@ -39,25 +43,25 @@ function Navbar({ isLogged, categories, user }) {
...
+              <Link to="/dashboard">لوحة التحكم</Link>
+            ) : (
+              <Link to="/my-appointment">مواعيدي</Link>
+            ),
         },
         {
-          label: (
-            <Button className="logout" onClick={logout}>LogOut</Button>
-          ),
+          label: 'خروج',
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/index.jsx:63]:
```suggestion
          label: 'تسجيل خروج',
```

File: client/src/Components/Navbar/index.jsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+          user={user}
+        />
+      </Drawer>
+    </Header>
+  );
+}
+
+Navbar.propTypes = {
+  isLogged: PropTypes.bool.isRequired,
+  user: PropTypes.arrayOf(PropTypes.oneOfType(PropTypes.object)).isRequired,
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/index.jsx:127]:
```suggestion
  user: PropTypes.arrayOf(PropTypes.oneOfType(PropTypes.object)),
```

File: client/src/Components/Navbar/index.jsx [hunk 9]
```diff
@@ -124,9 +122,13 @@ function Navbar({ isLogged, categories, user }) {
 
 Navbar.propTypes = {
   isLogged: PropTypes.bool.isRequired,
-  user: PropTypes.arrayOf(PropTypes.oneOfType(PropTypes.object)).isRequired,
+  user: PropTypes.objectOf(PropTypes.object()),
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/index.jsx:125]:
```suggestion
  user: PropTypes.shape({
    user: PropTypes.string.isRequired,
    role: PropTypes.string.isRequired,
  }),
```

File: client/src/Components/Navbar/RightMenu.jsx [hunk 10]
```diff
@@ -36,8 +41,12 @@ function RightMenu({ isLogged, mode, avatarMenu, user }) {
 RightMenu.propTypes = {
   mode: PropTypes.string.isRequired,
   isLogged: PropTypes.bool.isRequired,
-  user: PropTypes.arrayOf(PropTypes.oneOfType(PropTypes.object)).isRequired,
+  user: PropTypes.objectOf(PropTypes.object()),
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/RightMenu.jsx:44]:
```suggestion
  user: PropTypes.shape({
    user: PropTypes.string.isRequired,
    role: PropTypes.string.isRequired,
  }),
```

File: client/.eslintrc.json [hunk 11]
```diff
@@ -46,7 +46,7 @@
             "singleQuote": true
         }
         ],
-        "react/state-in-constructor": [0, "always"]
+        "react/state-in-constructor": [0, "always"],
```
Comment (Mu7ammadAbed) [inline client/.eslintrc.json:49]:
```suggestion
        "react/state-in-constructor": [0, "always"]
```

File: client/src/Components/Navbar/RightMenu.jsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+import PropTypes from 'prop-types';
+import { Menu, Button, Avatar, Dropdown, Space } from 'antd';
+import { UserOutlined, DownOutlined } from '@ant-design/icons';
+import { Link } from 'react-router-dom';
+
+function RightMenu({
+  isLogged,
+  mode,
+  avatarMenu,
+  user = { name: '', role: '' },
```
Comment (Mu7ammadAbed) [inline client/src/Components/Navbar/RightMenu.jsx:11]:
```suggestion
  user,
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Jehad91 — APPROVED** [review]

**Review by IsraaHamdi — APPROVED** [review]

**Pull request merged**